### PR TITLE
[runtime-security] replace overlay_numlower by an in_upper_layer boolean

### PR DESCRIPTION
**Pull request by lebauce**
### What does this PR do?

Replace overlay_numlower by an `is_upper_layer` boolean

### Motivation

The `overlay_numlower` integer attribute that was reported for files
and executables was unreliable. It was replaced by a simple boolean
attribute named `is_upper_layer` that is set to true when a file
is either only on the upper layer of an overlayfs filesystem, or
is an altered version of a file present in a base layer.

### CODE REVIEW

**Review by guedou — APPROVED** [review]
nit: I wonder if we could use a more meaningful attribute such as `is_base_image`.

**Comment by lebauce** [comment]
Indeed, originally we discussed about using is_base_image. There are 2 problems with it :
- "image" could be confusing : does it apply to virtual machine image ?
- is_base_image would be typically used using a "not" : "is the executable not part of the base image". If we are not using overlayfs, "not is_base_image" would always be true. "is_upper_layer" is only true when we use overlayfs and the file is part of the upper layer. Does it make sense ?

File: pkg/security/model/consts.go [hunk 1]
```diff
@@ -280,6 +280,12 @@ var (
 	kernelCapabilitiesStrings = map[int]string{}
 )
 
+// File flags
+const (
+	LowerLayer = 1 << 0
```
Comment (safchain) [inline pkg/security/model/consts.go:285]: nit: iota + 1

File: pkg/security/probe/model.go [hunk 2]
```diff
@@ -382,52 +402,62 @@ func (ev *Event) ResolveCredentialsCapPermitted(e *model.Credentials) int {
...
 // ResolveSetgidFSGroup resolves the file-system group of the Setgid event
 func (ev *Event) ResolveSetgidFSGroup(e *model.SetgidEvent) string {
-	if len(e.FSGroup) == 0 {
+	if len(e.FSGroup) == 0 && ev != nil {
 		e.FSGroup, _ = ev.resolvers.UserGroupResolver.ResolveUser(int(e.FSGID))
 	}
 	return e.FSGroup
 }
 
+// ResolveExecInUpperLayer resolves whether the file is in an upper layer
```
Comment (safchain) [inline pkg/security/probe/model.go:451]: doesn't seem to be used ?

Comment (lebauce) [reply pkg/security/probe/model.go:451]: Indeed. Removed

**Pull request merged**